server/src/localization.cpp

```diff
@@ -3,17 +3,94 @@
 //
 
 #include "localization.hpp"
+#include "data_store.hpp"
+
+extern insNode_t * insNoderoot;
 
 namespace ins_service
 {
 
+wifiParams_t * Localization::FillNodeDataPoints(wifiParams_t *  wifiNodeBlock, AccessPointRssiListPair mac_rssi_)
+{
+	initKalmanParams(wifiNodeBlock);
+
+	wifiNodeBlock->macAddress = (char *) mac_rssi_.first.mac_addr.c_str();
+
+	wifiNodeBlock->noSampleData = mac_rssi_.second.size();
+
+	for (int i = 0; i < mac_rssi_.second.size(); ++i)
+	{
+		wifiNodeBlock->rssisampledata[(i%NUMBER_SAMPLES)] = (float)mac_rssi_.second[i];
+	}
+
+	loadLCFGParams(wifiNodeBlock);
+
+	return wifiNodeBlock;
+}
+
+
+insNode_t * Localization::FillNodesDataPoints(const char * device_id, std::vector<AccessPointRssiListPair> mac_rssi_list)
+{
+	insNode_t * insNode;
+
+	if ( (insNode = findWifiNode(insNoderoot, device_id)) == NULL)
+	{
+		insNode =  createInsNodeListDevice(device_id); // check to make sure nodeblock exits!!
+	}
+
+	std::string msg = "Number of distinct mac addresses received from device: " + std::string(device_id) + " is: " + std::to_string(mac_rssi_list.size()) + " with: " +  std::to_string(mac_rssi_list[0].second.size()) + " data points.";
+	console_->debug(msg);
+
+	for (int i = 0; i < mac_rssi_list.size(); ++i)
```

Comment:
Iterate through the vector instead such as:
`for (auto mac_rssi : mac_rssi_list)`

Reply:
Yeh, I thought so too. But when thing were not working, I went low level. :smile: